### PR TITLE
Make the websocket connection auto reconnect even on server restart

### PR DESCRIPTION
**Pull request by prashnts**
Hi! Here's a small patch to make websocket connections a bit more robust.

Before: If I restart my emulator process, I need to refresh the webpage (the retries did not work for me, it never reconnected).

Now: If I restart my emulator, the websocket connection is automatically re-established, and I do not need to refresh.

### CODE REVIEW

File: RGBMatrixEmulator/adapters/browser_adapter/static/assets/client.js [hunk 1]
```diff
@@ -15,38 +14,11 @@ function init() {
...
-    function requestImage() { 
+    function requestImage() {
         requestStartTime = performance.now();
-
-        waitForSocketReady(socket, function () {
-            socket.send('more');
-        });
-    }
-
-    function waitForSocketReady(socket, callback) {
```
Comment (prashnts) [inline RGBMatrixEmulator/adapters/browser_adapter/static/assets/client.js:29]: We no longer need to check for socket status as once we get an error we recreate the connection.

However this may not be desirable for allowing a slow websocket server to recover. (it will always be reconnected).

File: RGBMatrixEmulator/adapters/browser_adapter/static/assets/client.js [hunk 2]
```diff
@@ -64,25 +36,33 @@ function init() {
...
+            // Handle retries by recreating the connection to websocket.
+            console.warn(`RGBME WebSocket connection lost. Retrying in ${WS_RETRY_DELAY / 1000}s.`)
+            setTimeout(function() {
+                // We generate socket with a timeout to make sure server has time to recover.
+                socket = generateSocket();
+            }, 2000);
+        }
+
+        ws.onerror = function() {
+            ws.close();
```
Comment (prashnts) [inline RGBMatrixEmulator/adapters/browser_adapter/static/assets/client.js:56]: The `close` method is important to not cause memory leaks.

Comment (prashnts) [reply RGBMatrixEmulator/adapters/browser_adapter/static/assets/client.js:56]: And apparently this might be not enough according to this SO answer. I'll check. In my tests I had no issues with it. 

https://stackoverflow.com/a/11982071

Comment (ty-porter) [reply RGBMatrixEmulator/adapters/browser_adapter/static/assets/client.js:56]: I think if we run into these edge cases in the future we can optimize it then

Most people use the emulator as a dev tool so it's on their local machine and short-lived.

**Review by ty-porter — APPROVED** [review]
Great contribution, thank you. I've meant to look at this but it has never been high on my list of priorities.
